api/src/services/payload.ts

```diff
@@ -115,6 +116,19 @@ export class PayloadService {
 			if (Array.isArray(value)) return value.join(',');
 			return value;
 		},
+		async 'storage-public-url'({ action, payload }) {
+			if (action !== 'read') return;
+			const storage_config = getConfigFromEnv(`STORAGE_${payload.storage.toUpperCase()}_`);
+			if (!storage_config.publicUrl) return `${process.env.PUBLIC_URL}/assets/${payload.id}`;
+			try {
+				const urlTemplate = template(storage_config.publicUrl, {
```

Comment:
We have micromustache available for this exact templating

Reply:
Switched to use micromustache.